### ISSUE TITLE
Support copying data between inherited types

### ISSUE THREAD

**Issue by talon55** [danielnorberg/auto-matter#70]
Currently value-type inheritance works great, but there's no good way to copy data between an instance of a value type to one that inherits from it.

For example:
```java
@AutoMatter
public interface FooValue {
	String foo();
}

@AutoMatter
public interface BarValue extends FooValue {
	String bar();
}

// If we have...
FooValue foo = new FooValueBuilder()
	.foo("Hello ")
	.build();

// It would be great to be able to do this:
BarValue bar = BarValueBuilder.from(foo)
	.bar("world!")
	.build();

// Or maybe this, to support inheriting from more than one interface:
BarValue bar2 = new BarValueBuilder()
	.include(foo) // better names could be ".import()" or ".copy()"
	.bar("world!")
	.build();
```

This could probably be simplified if you only include support for ancestor interfaces which are also annotated with `@AutoMatter`.

The above example is a bit contrived, of course, but you could see how this would get to be a huge pain if the base interface had a lot of fields.

**Comment by danielnorberg** [comment]
This would indeed be very useful and might be straightforward to implement. I did recently observe a case where the runtime mapping solution no longer worked. I’ll try to implement this soon.

**Comment by danielnorberg** [comment]
Released in `0.17.0`, please try it out.
